process/roundActivation/roundActivation.go

```diff
@@ -39,6 +40,22 @@ func (ra *roundActivation) IsEnabled(name string, round uint64) bool {
 	return false
 }
 
+// IsEnabled checks if the queried round flag name is enabled in the current processed round
+func (ra *roundActivation) IsEnabled(name string) bool {
+	ra.mutex.RLock()
+	currRound := ra.round
+	ra.mutex.RUnlock()
+
+	return ra.IsEnabledInRound(name, currRound)
```

Comment:
I would do this a little bit different in order to minimize the usage of the mutex.
I would create an unexported function called `isEnabledInRound` that do not acquire the mutex
The exported function `IsEnabledInRound` calls the `isEnabledInRound` under mutex r-locking. 
The `IsEnabled` fetches & calls `isEnabledInRound` under the same mutex r-locking code block.

Reply:
Right, good suggestion and explanation. :+1: 
I've applied it.